src/osg_display/transfer_datasource.py

```diff
@@ -124,26 +123,26 @@ def load_cached(self):
                      remove_data.append(time)
             for time in remove_data:
                 del self.data[time]
-        except Exception, e:
+        except Exception as e:
             log.warning("Unable to load cache; it may not exist. Error: %s" % \
                str(e))
 
     def save_cache(self):
         now = datetime.datetime.now()
         old_keys = []
-        for key in self.data.keys():
+        for key in list(self.data.keys()):
```

Comment:
```suggestion
        for key in self.data.keys():
```

list conversion is unnecessary here